WcaOnRails/app/webpacker/components/Competitions/index.jsx

```diff
@@ -0,0 +1,215 @@
+              {(permissions.can_administer_competition === '*' || permissions.can_administer_competition.includes(competition.id)) && (
+                <>
+                  <Popup
+                    content="View the Delegate Report"
+                    trigger={(
+                      <a href={competitionReportUrl(competition.id)}>
+                        <Icon name="file alternate" />
+                      </a>
+                    )}
+                  />
+                  <Popup
+                    content="Edit the Report"
+                    trigger={(
+                      <a href={competitionReportEditUrl(competition.id)}>
+                        <Icon name="edit" />
+                      </a>
+                    )}
+                  />
+                </>
+              )}
+            </Table.Cell>
+          </Table.Row>
+        ))}
+      </TableBody>
+    </Table>
+  );
+}
+
+function PastCompetitionsTable({ competitions, permissions }) {
+  return (
+    <Table striped>
+      <TableHeader>
+        <Table.Row>
+          <Table.HeaderCell>
+            {I18n.t('competitions.adjacent_competitions.name')}
+          </Table.HeaderCell>
+          <Table.HeaderCell>
+            {I18n.t('competitions.adjacent_competitions.location')}
+          </Table.HeaderCell>
+          <Table.HeaderCell>
+            {I18n.t('competitions.adjacent_competitions.date')}
+          </Table.HeaderCell>
+          <Table.HeaderCell />
+          <Table.HeaderCell />
+        </Table.Row>
+
+      </TableHeader>
+      <TableBody>
+        {competitions.map((competition) => (
+          <Table.Row>
+            <Table.Cell>
+              {competition.name}
+            </Table.Cell>
+            <Table.Cell>
+              {competition.location}
+            </Table.Cell>
+            <Table.Cell>
+              {competition.date}
+            </Table.Cell>
+            <Table.Cell>
+              {!competition['results_posted?'] && <Icon name="calendar" />}
+            </Table.Cell>
+            <Table.Cell>
+              {competition['results_posted?'] && <Icon name="check circle" />}
+            </Table.Cell>
+            <Table.Cell>
+              {(permissions.can_administer_competition === '*' || permissions.can_administer_competition.includes(competition.id)) && (
+                <>
+                  <Popup
+                    content="View the Delegate Report"
+                    trigger={(
+                      <a href={competitionReportUrl(competition.id)}>
+                        <Icon name="file alternate" />
+                      </a>
+                    )}
+                  />
+                  <Popup
+                    content="Edit the Report"
+                    trigger={(
+                      <a href={competitionReportEditUrl(competition.id)}>
+                        <Icon name="edit" />
+                      </a>
+                    )}
+                  />
+                </>
```

Comment:
The two tables have a lot of repeated code, which is understandable. I think big chunks like this which are identical should be factored out into reusable components.